chainio/txmgr/txmgr.go

```diff
@@ -82,19 +82,21 @@ func NewSimpleTxManager(
 func (m *SimpleTxManager) Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error) {
 
 	// Estimate gas and nonce
-	m.log.Debug("Estimating gas and nonce", "tx", tx.Hash().Hex())
+	// can't print tx hash in logs because the tx changes below when we complete and sign it
+	// so the txHash is meaningless at this point
+	m.log.Debug("Estimating gas and nonce")
```

Comment:
Assuming there can be many calls to Send in parallel on different goroutines, how the debug log records without the transaction identifier will help in debugging?

Reply:
that's a good point. @samlaf should we keep the hash - and saying - it's an unsigned hash to make it clear?

Reply:
the problem is we'd want some kind of call tracing, but we if use the txhash inside and txreceipt.tx.hash outside, then the tracing "identifier" is not even unique..

Reply:
I would say let's cross that bridge when it's needed only. For now we won't be managing multiple wallets, so assuming a single tx is in flight at any point in time is safe I would think.

Reply:
maybe eventually we wrap a transaction lifecyle with a unique identifier - that would be ideal to log and can work with multiple go routines, gas adjustments. it's okay for now